src/WebSocketSubscriptionServer.ts

```diff
@@ -0,0 +1,170 @@
+/* eslint-disable max-classes-per-file */
+import { EventEmitter } from 'events';
+import type * as http from 'http';
+import url from 'url';
+
+import ws from 'ws';
+
+import SubscriptionServer, {
+  SubscriptionServerConfig,
+} from './SubscriptionServer';
+import { MessageType } from './types';
+
+interface Message {
+  type: MessageType;
+  payload: any;
+  ackId?: number;
+}
+
+class GraphQLSocket extends EventEmitter {
+  protocol: 'graphql-transport-ws' | 'socket-io';
```

Comment:
The plan is to support: [GraphQL over websockets protocol](https://github.com/enisdenjo/graphql-ws/blob/master/PROTOCOL.md) next

Reply:
i am a little confused, this `GraphQLSocket` only seems to be used by the websocket, so why have the socket-io protocol as an allowed value?

Reply:
yeah sorry. So in this case "socket-io" is really "4C's bespoke protocol of connect and authenticate calls" I don't remember why i named it "socket-io" maybe just because it depends on the socket-io handshake events

Reply:
and by "I named it" what do you mean by that? where is this name specified? do you think it's too late to change? maybe we can add a comment to avoid confusion

Reply:
it's made up by me. this theoretically supposed to match the web socket sub protocol. 'socket-io' is a subprotocol used by socket-io which is why i just named it that for the pure web socket but it's not actually that protocol so i should disambiguate and name it `4c-subscription-server`